crates/wysiwyg/src/composer_model/delete_text.rs

```diff
@@ -92,9 +93,40 @@ where
         self.do_replace_text_in(S::default(), start, end)
     }
 
+    /// If we have cursor at the edge of or inside a non-editable text node, expand the selection to cover
+    /// the whole of that node before continuing with the backspace/deletion flow
+    fn handle_non_editable_selection(&mut self) {
+        let (s, e) = self.safe_selection();
+        let range = self.state.dom.find_range(s, e);
+
+        let first_leaf = range.locations.iter().find(|loc| {
+            loc.is_leaf() || (loc.kind.is_block_kind() && loc.is_empty())
+        });
+        if let Some(leaf) = first_leaf {
+            let parent_link_loc =
+                range.deepest_node_of_kind(Link, Some(&leaf.node_handle));
+            if let Some(link) = parent_link_loc {
+                if self
+                    .state
+                    .dom
+                    .lookup_container(&link.node_handle)
+                    .is_immutable_link()
+                {
+                    self.select(
+                        Location::from(link.position),
+                        Location::from(link.position + link.length),
+                    );
+                }
+            }
+        }
+    }
```

Comment:
I feel like this is pretty clunky but wanted to get it working first. If this approach should be changed or the code moved to a different part of the codebase, let me know. Haven't done rust stuff for a little while.